packages/logger/src/Logger.ts

```diff
@@ -292,9 +292,8 @@ class Logger extends Utility implements ClassThatLogs {
         }
 
         this.addContext(context);
-        if (options) {
-          this.logEventIfEnabled(event, options.logEvent);
-        }
+        const logEvent = options ? options.hasOwnProperty('logEvent') ? options.logEvent : undefined : undefined;
+        this.logEventIfEnabled(event, logEvent);
```

Comment:
(nit) I find this difficult to read. Will this be better?

```
const shouldLogEvent = undefined;
if( options && options.hasOwnProperty('logEvent') ) {
  shouldLogEvent = options.logEvent;
}
this.logEventIfEnabled(event, shouldLogEvent);
```

Or we can wait until we drop Node12 support and simply use `options?.logEvent`?

Reply:
Changed to your suggestion, only difference is that I made `shouldLogEvent` a `let` instead of `const`